packages/x-date-pickers/src/internals/hooks/useViews.tsx

```diff
@@ -190,23 +191,20 @@ export function useViews<TValue, TView extends DateOrTimeViewWithMeridiem>({
       const globalSelectionState =
         isSelectionFinishedOnCurrentView && hasMoreViews ? 'partial' : currentViewSelectionState;
 
-      onChange(value, globalSelectionState);
-      if (isSelectionFinishedOnCurrentView) {
+      onChange(value, globalSelectionState, selectedView);
+      // detect out of order selection
+      if (selectedView && selectedView !== view) {
```

Comment:
Handling out-of-order selections, like `DesktopDateTimePicker` and `MultiSectionDigitalClock`.

Reply:
"out of order" is not super clear if we don't have context. From what understand, it fix the special case where we can see multiple view together and so the clicked one (`selectedView`) can a different one from the one with focus (`view`)

What about 
```
// Detects if the selected view is not the active one.
// Can happen if multiple views are displayed like for `DesktopDateTimePicker` or `MultiSectionDigitalClock`.
```